src/client.cpp

```diff
@@ -120,8 +120,8 @@ ADDON_STATUS ADDON_GetStatus()
   CLockObject lock(g_mutex);
 
   // Check that we're still connected
-  if (m_CurStatus == ADDON_STATUS_OK && !tvh->IsConnected())
-    m_CurStatus = ADDON_STATUS_LOST_CONNECTION;
+  if (m_CurStatus == ADDON_STATUS_OK && !tvh->IsSuspended() && !tvh->IsConnected())
+    return ADDON_STATUS_LOST_CONNECTION;
```

Comment:
promoting this function upper layers (generalization) is wrong here. even simple client has no server connection. from kodi POV "connection lost" is not a well defined status. this was a quick hack that never should have been added as a states.
you could add some method to PVR API that allows to signal state and have pvr component handle it. kodi's general addon loader should not be aware of addon specific states.